std_execution.bs

```diff
@@ -1247,6 +1248,133 @@ talk about the semantics of some of the algorithms proposed in this paper.
+    making forward progress.
+
+      The risk of deadlock might occur, for example, if a thread-pool with a
+    small number of threads is executing code that creates a sender representing
+    an eagerly-executing operation and then calls the destructor of that sender
+    without joining it (e.g. because an exception was thrown). If the current
+    thread blocks waiting for that eager operation to complete and that eager
+    operation cannot complete until some entry enqueued to the thread-pool's
+    queue of work is run then the thread may wait for an indefinite amount of
+    time. If all thread of the thread-pool are simultaneously performing such
+    blocking operations then deadlock can result.
+
+There are also minor variations on each of these choices. For example:
+
+4.  A variation of (1): Call `std::terminate` if an eager sender is destructed
+    without joining it. This is the approach that `std::thread` destructor
+    takes.
+
+5.  A variation of (2): Request cancellation of the operation before detaching.
+    This reduces the chances of operations continuing to run indefinitely in the
+    background once they have been detached but does not solve the
+    lifetime- or shutdown-related challenges.
+
+6.  A variation of (3): Request cancellation of the operation before blocking on
+    its completion. This is the strategy that `std::jthread` uses for its
+    destructor. It reduces the risk of deadlock but does not eliminate it.
+
+### Eager senders complicate algorithm implementations ### {#design-lazy-algorithms-complexity}
+
+Algorithms that can assume they are operating on senders with strictly lazy
+semantics are able to make certain optimizations that are not available if
+senders can be potentially eager. With lazy senders, an algorithm can safely
+assume that a call to `execution::start` on an operation state strictly happens
+before any receiver methods are called. This frees the algorithm from needing to
+resolve potential race conditions between the start and completion of the work.
+
+When an algorithm needs to deal with potentially eager senders, the potential
+race conditions can be resolved one of several ways, none of which is desirable:
+
+1.  Assume the worst and implement the algorithm defensively, assuming all
+    senders are eager. This obviously has overheads both at runtime and in
+    algorithm complexity. Resolving race conditions is hard.
+
+2.  Require senders to declare whether they are eager or not with a query.
+    Algorithms can then implement two different implementation strategies, one
+    for strictly lazy senders and one for potentially eager senders. This
+    addresses the performance problem of (1) while compounding the complexity
+    problem.
+
+3.  Push the problem back onto eager senders by requiring them to never call a
+    receiver's `set_*` completion until after `execution::start` has been called
+    on the operation state. That puts the onus of dealing with the race
```

Comment:
Again, this is something that is already required as part of the sender/receiver contract.

The limitation on algorithms is that it "should not do the work" until the operation is started, not that it "should not complete" until it is started.